fints/client.py

```diff
@@ -59,18 +63,29 @@ def get_statement(self, account, start_date, end_date):
         dialog.sync()
         dialog.init()
 
-        msg = self._create_statement_message(dialog, account, start_date, end_date, None)
-        logger.debug('Send message: {}'.format(msg))
+        def _get_msg():
+            return self._create_statement_message(dialog, account, start_date, end_date, None)
+
+        with self.pin.protect():
+            logger.debug('Send message: {}'.format(_get_msg()))
+
+        msg = _get_msg()
         resp = dialog.send(msg)
         touchdowns = resp.get_touchdowns(msg)
         responses = [resp]
         touchdown_counter = 1
 
         while HKKAZ.type in touchdowns:
             logger.info('Fetching more results ({})...'.format(touchdown_counter))
-            msg = self._create_statement_message(dialog, account, start_date, end_date, touchdowns[HKKAZ.type])
-            logger.debug('Send message: {}'.format(msg))
 
+            # TODO: do not create functions in a loop
```

Comment:
I think it might be okay here to just have the call to ``_create_statement_message`` twice instead of creating the inline function. Yes, it is a new source of errors, but I think it is okay here.